File: de.wwu.md2.android.lib/src/main/java/de/uni_muenster/wi/md2library/model/type/interfaces/Md2NumericType.java
package de.uni_muenster.wi.md2library.model.type.interfaces;

/**
 * {@inheritDoc}
 * <p/>
 * Representation of numeric types in MD2-DSL
 * <p/>
 * Created on 07/07/2015
 *
 * @author Fabian Wrede
 * @version 1.0
 * @since 1.0
 */
public interface Md2NumericType extends Md2DataType {
    /**
     * greater than
     *
     * @param value the value
     * @return the boolean
     */
    boolean gt(Md2NumericType value);

    /**
     * greater than or equal
     *
     * @param value the value
     * @return the boolean
     */
    boolean gte(Md2NumericType value);

    /**
     * less than
     *
     * @param value the value
     * @return the boolean
     */
    boolean lt(Md2NumericType value);

    /**
     * less than or equal
     *
     * @param value the value
     * @return the boolean
     */
    boolean lte(Md2NumericType value);
}

File: de.wwu.md2.framework/res/resources/ios/lib/model/type/MD2NumericType.swift
//
//  MD2NumericType.swift
//  md2-ios-library
//
//  Created by Christoph Rieger on 21.07.15.
//  Copyright (c) 2015 Christoph Rieger. All rights reserved.
//

protocol MD2NumericType: MD2DataType {
    
    /**
        Greater-than comparison method.
    
        :param: value The value to compare with.
    
        :returns: Whether the object is greater than the parameter value.
    */
    func gt(value: MD2NumericType) -> Bool
    
    /**
        Greater-or-equal comparison method.
    
        :param: value The value to compare with.
    
        :returns: Whether the object is greater or equal to the parameter value.
    */
    func gte(value: MD2NumericType) -> Bool
    
    /**
        Lower-than comparison method.
    
        :param: value The value to compare with.
    
        :returns: Whether the object is lower than the parameter value.
    */
    func lt(value: MD2NumericType) -> Bool
    
    /**
        Lower-or-equal comparison method.
    
        :param: value The value to compare with.
    
        :returns: Whether the object is lower or equal to the parameter value.
    */
    func lte(value: MD2NumericType) -> Bool

}